### PR TITLE
Fix get give

### PR DESCRIPTION
**Pull request by aszlam**
# Description

- updates Get/Give/Bring/Drop in interpreter (which was nonfunctional); 
- makes a test for these, 
- partially harmonizes pyworld and cuberite backends to allow these
- uses backend flag to handle differences between the two settings

## Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [x] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [x] I want a high level review. 
- [ ] I want a deep design review.

# Testing

Tested in MC, and wrote new tests in pyworld

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [x] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

# Notes:
in the near future, FakeAgent needs to be re-written to better integrated with pyworld.

### CODE REVIEW

File: droidlet/interpreter/craftassist/mc_interpreter.py [hunk 1]
```diff
@@ -496,63 +495,89 @@ def new_tasks():
...
+                    # this is bring to an entity; but for now will just drop it at entities location
+                    mems = self.subinterpret["reference_locations"](self, speaker, d_receiver)
+                else:  # d_receiver.get("location"):
+                    assert d_receiver.get("location") is not None
+                    mems = self.subinterpret["reference_locations"](
+                        self, speaker, d_receiver["location"]
+                    )
+            except NextDialogueStep:
+                # TODO allow for clarification
+                raise ErrorWithResponse("I don't understand where you want me to move.")
```
Comment (kavyasrinet) [inline droidlet/interpreter/craftassist/mc_interpreter.py:569]: nit: more informative error response: "I don't understand where you want me to place the item."

File: droidlet/interpreter/craftassist/mc_interpreter.py [hunk 2]
```diff
@@ -496,63 +495,89 @@ def new_tasks():
...
+            # FIXME this should go in the ref_location subinterpret:
+            steps, reldir = interpret_relative_direction(self, d_receiver)
+            pos, _ = self.subinterpret["specify_locations"](self, speaker, mems, steps, reldir)
+            # TODO: can this actually happen?
+            if pos is None:
+                raise ErrorWithResponse("I don't understand where you want me to move.")
```
Comment (kavyasrinet) [inline droidlet/interpreter/craftassist/mc_interpreter.py:575]: same^

File: droidlet/interpreter/craftassist/tasks.py [hunk 3]
```diff
@@ -961,35 +961,52 @@ class Get(Task):
...
         super().__init__(agent)
-        self.idm = task_data.get("idm")
         self.pos = task_data["pos"]
         self.eid = task_data["eid"]
         self.obj_memid = task_data["obj_memid"]
         self.approx = 1
-        self.attempts = 10
-        if self.idm is not None:
+        self.attempts = 20
+        self.type_name = agent.memory.get_mem_by_id(self.obj_memid).type_name
```
Comment (kavyasrinet) [inline droidlet/interpreter/craftassist/tasks.py:975]: what does the type_name field represent ?

File: droidlet/interpreter/craftassist/tasks.py [hunk 4]
```diff
@@ -1067,25 +1082,35 @@ def step(self):
...
+            return
+
         node = agent.memory.get_mem_by_id(self.obj_memid)
         count = node.count
-        id, m = self.idm
-        agent.drop_inventory_item_stack(id, m, count)
-        ItemStackNode.remove_from_inventory(agent.memory, node)
-
-        mindist, dropped_item_stack = self.get_nearby_new_item_stack(agent, id, m)
+        if getattr(agent, "backend", "cuberite") == "cuberite":
```
Comment (kavyasrinet) [inline droidlet/interpreter/craftassist/tasks.py:1095]: oooo this should go away once we fix the integration with pyworld right ?

**Review by kavyasrinet — APPROVED** [review]

File: droidlet/interpreter/craftassist/tasks.py (hunk 3 above)
Comment (aszlam) [reply droidlet/interpreter/craftassist/tasks.py:975]: the name of the object being dropped, e.g. "stone".

File: droidlet/interpreter/craftassist/tasks.py (hunk 4 above)
Comment (aszlam) [reply droidlet/interpreter/craftassist/tasks.py:1095]: probably not- getting and dropping things is different in pyworld (where you explicitly pick) than cuberite (where walking close picks it up); and cuberite tracks item stacks in a different way well

**Pull request merged**